src/BoundaryConditions/flux_calculations.jl

```diff
@@ -1,61 +1,130 @@
 end
 
 """
-    get_boundary_flux(model, bc::DragLawCondition, uv, Y, Ya)
+    get_boundary_flux(model, ::NoVectorFlux, var::Fields.Field, Y, Ya)
 
-    Vertical momentum fluxes for a bulk-formulation drag law, given momentum exchange coefficients
+    Vertical flux assignment for an AxisTensor with CovariantAxis{(1,2)} in no-flux conditions (i.e., insulating wall)
 """
-function get_boundary_flux(model, bc::DragLawCondition, uv, Y, Ya)
+function get_boundary_flux(model, ::NoVectorFlux, var::Fields.Field, Y, Ya)
     FT = eltype(Y)
-    coefficients =
-        eltype(bc.coefficients) == FT ? bc.coefficients : bc.coefficients(Y, Ya)
+    flux =
+        Geometry.Covariant3Vector(FT(0)) ⊗
+        Geometry.Covariant12Vector(FT(0), FT(0))
+end
 
-    uv_1 = first_interior(uv)
+"""
+    get_boundary_flux(model, bc::DragLaw, uv, Y, Ya)
+
+    Vertical momentum fluxes for a bulk-formulation drag law, given the momentum exchange coefficient
+"""
+function get_boundary_flux(model, bc::DragLaw, uv, Y, Ya)
+    FT = eltype(Y)
+    coefficient =
+        eltype(bc.coefficient) == FT ? bc.coefficient : bc.coefficient(Y, Ya)
+
+    uv_1 = Fields.level(uv, 1)
     u_wind = norm(uv_1)
 
-    flux = Geometry.WVector(coefficients.Cd * u_wind) ⊗ uv_1
+    flux =
+        Geometry.Contravariant3Vector(coefficient * u_wind) ⊗
+        Geometry.Covariant12Vector(FT(0), FT(0)) #uv_1 # change when ClimaCore enables Fields for specifying boundary conditions (#325)
 end
 
 """
     get_boundary_flux(
         model,
-        bc::BulkFormulaCondition,
+        bc::BulkFormula,
         ρc::Fields.Field,
         Y,
         Ya,
     )
 
-    Vertical fluxes for arbitrary variables with a bulk-formulation drag law, 
+    Vertical fluxes for arbitrary variables with the bulk aerodynamic turbulent formula, 
     given variable exchange coefficients. (e.g. for energy, or tracers)
 """
-function get_boundary_flux(
-    model,
-    bc::BulkFormulaCondition,
-    ρc::Fields.Field,
-    Y,
-    Ya,
-)
+
+function get_boundary_flux(model, bc::BulkFormulaDryTotalEnergy, ρc::Fields.Field, Y, Ya)
     FT = eltype(Y.base)
-    coefficients =
-        eltype(bc.coefficients) == FT ? bc.coefficients : bc.coefficients(Y, Ya)
-    c_sfc = bc.surface_field
 
-    ρ_1 = first_interior(Y.base.ρ)
-    ρc_1 = first_interior(ρc)
-    uv_1 = first_interior(Y.base.uv)
-    u_wind = norm(uv_1)
+    sfc = bc.surface_fields
 
-    flux =
-        Geometry.WVector(coefficients.Ch * u_wind * ρ_1 * (ρc_1 / ρ_1 - c_sfc))
+    # get relevant states of the first atmos internal point and the surface 
+    p = calculate_pressure(
+        Y,
+        Ya,
+        model.base,
+        model.thermodynamics,
+        model.moisture,
+        model.parameters,
+        FT,
+    )
+
+    qt_sfc = FT(0)
+    cv_d = TD.cv_m(model.parameters, FT)
+
+    # internal energies
+    e_int_in = Fields.level(TD.air_temperature_from_ideal_gas_law.(model.parameters, p, Y.base.ρ) .* cv_d, 1)
```

Comment:
is there a reason not to use TD. internal_energy(ts) here? 
https://github.com/CliMA/Thermodynamics.jl/blob/119e7271497b06cdeffaf86112c475c76ec67e5b/src/relations.jl#L377